### DISCUSSION TITLE
Spawning old items at start of new round can trigger OnInserted SE

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15372] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

different from spawning new items inside item container
these items were already here and should not trigger OnInserted SE

### Reproduction steps

.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
We were implementing a fix to this, but started wondering how it should actually work: should it trigger when an item is placed into a container for the first time no matter how it's being placed (should e.g. initial supplies spawning in the sub or items placed into character inventories on spawn trigger it?), or only when a player is manually putting something in an inventory?

**Reply by A2ur3** [reply]
>  only when a player is manually putting something in an inventory

does it mean items spawned by status effect won't trigger OnInserted anymore?
if yes I'd prefer the former although in some situations it looks weird (magazine will trigger status effect when put inside gun and character spawns with a gun and magazine)

**Reply by Regalis11** [reply]
I suppose they could also trigger when inserted by a StatusEffect. I guess one way this could work is that it triggers on any item that gets put inside a container mid-round (no matter if it's put by a character, status effect or a respawn?), when not in a loading screen.

**Reply by A2ur3** [reply]
if a midround respawn can still trigger OnInserted, it can't fully address the issue
think about a item that has a hidden inventory where you can insert up to 4 alien circuitries and every insert decreases 25% item condition
so you can either insert 4 in one go or insert 1 circuitry and wait for 3 respawn

**Reply by Regalis11** [reply]
Thanks for the example! The behavior I ended up going with for now is that the effect executes when the item is inserted into the container for the first time (not when loading a sub or a character where the item is already in the container).

In other words:
- Executes when a character, status effect or anything else puts the item in the container mid-round.
- Executes when the is first place in the container - e.g. when the initial supplies in a submarine are generated or when a character spawns for the first time.
- Does not execute when you load an existing save
- Does not execute when a character who's previously spawned spawns again

Does it sound like this behavior would work for your use case?

**Comment by Regalis11** [comment]
Fixed in our private repo, fix coming up in a future build.
